test/test_parsers.py

```diff
@@ -10,3 +11,39 @@ def test_whatsapp():
         assert row_res.equals(row_target), row_res.compare(
             row_target, result_names=("result", "target")
         )
+
+
+def test_facebookMessenger():
+    parser = FacebookMessengerParser("test/facebookMessenger/testlog.json")
+    parser.parse_file_into_df()
+    # set convert_dates to false since if set to true, the conversion will use UTC 0 instead of using the users local timezone
+    df_test = pd.read_json(
+        "test/facebookMessenger/target.json", orient="records", convert_dates=False
+    )
+    adjustUtcTimestamp(df_test)
+    for (_, row_res), (_, row_target) in zip(parser.df.iterrows(), df_test.iterrows()):
+        assert row_res.equals(row_target), row_res.compare(
+            row_target, result_names=("result", "target")
+        )
+
+
+def adjustUtcTimestamp(df: pd.DataFrame):
+    for index in df.index:
+        # Convert timestamp to datetime format with the local timezone
+        df.at[index, "datetime"] = datetime.datetime.fromtimestamp(
+            df.at[index, "datetime"] / 1000
+        )
+        # Adjust the hour by the UTC hour difference. i.e. If hour is 15 and the timezone is UTC-6, then hour will be 9
+        df.at[index, "hour"] = df.at[index, "hour"] + (
+            round(
+                (
+                    round(
+                        (
+                            datetime.datetime.now() - datetime.datetime.utcnow()
+                        ).total_seconds()
+                    )
+                    / 1800
+                )
+                / 2
+            )
+        )
```

Comment:
Why do we even need to accommodate for timezone differences?
The timestamps both come from textfiles, where is the localization added to them?
I think if we stick to UTC throughout the test, we should be fine, right?

Reply:
I thought creating these tests would be as easy as the WhatsApp ones, but apparently not. In the facebookParser, the datetime string is converted to a readable date like this
```       
parsed_message = {
            "datetime": datetime.datetime.fromtimestamp(mess["timestamp_ms"] / 1000),
            "author": mess["sender_name"],
            "message": body,
        }
```
It seems that datetime.fromtimestamp uses UTC0 to convert a timestamp to a readable form. However, when we use `pd.read_json` for reading the target.json in the test_parses.py, it seems that it takes the user's timezone to convert the timestamp to readable form.

That was my theory and why I had to do it this way. You could test it out as well by modifying the test_facebookMessenger as follows: 
```
def test_facebookMessenger():
    parser = FacebookMessengerParser("test/facebookMessenger/testlog.json")
    parser.parse_file_into_df()

    df_test = pd.read_json(
        "test/facebookMessenger/target.json", orient="records"
    )

    for (_, row_res), (_, row_target) in zip(parser.df.iterrows(), df_test.iterrows()):
        assert row_res.equals(row_target), row_res.compare(
            row_target, result_names=("result", "target")
        )
```

Reply:
I see. Did you try setting `convert_dates = True` and converting the timestamp to UTC via [tz_localize](https://pandas.pydata.org/docs/reference/api/pandas.Series.tz_localize.html)? I think this would be much cleaner

Reply:
I only tried with `onvert_dates = True`, i'll test it out with tz_localize later on the day. Will let you know the result